### ISSUE TITLE
Confusion about the model

### ISSUE THREAD

**Issue by Valhir924** [heshuting555/DsHmp#21]
Thanks for your impressive works.
When I'm reading your works, I encounter a puzzle which I fail to solve it out myself. Would you be so kind to help me out?
In section 3.2 Hierarchical Motion Perception, it mentions that "M is a token merging operation that blends two tokens of neighboring frames into a single one using a weighted average based on ˆAh." 
However, I only find lines below in your codes:

def pairwise_average(self, tensor):
        if tensor.shape[1] % 2 != 0:
            tensor = torch.cat([tensor, tensor.narrow(1, -1, 1)], dim=1)
        tensor = (tensor[:, ::2] + tensor[:, 1::2]) / 2
        return tensor

I'm not sure whether function pairwise_average is this the merge operation.
